gui/wxpython/datacatalog/tree.py

```diff
@@ -888,38 +933,44 @@ def OnEndDrag(self, node, event):
             Debug.msg(1, "DROP DONE")
             self.OnPasteMap(event)
 
-    def OnSwitchLocationMapset(self, event):
+    def OnSwitchDbLocationMapset(self, event):
         genv = gisenv()
         if self.selected_location[0].label == genv['LOCATION_NAME']:
             self.changeMapset.emit(mapset=self.selected_mapset[0].label)
+        elif self.selected_grassdb[0].label == genv['GISDBASE']:
+            self.changeLocation.emit(mapset=self.selected_mapset[0].label,
+                                     location=self.selected_location[0].label)
         else:
-            self.changeLocation.emit(mapset=self.selected_mapset[0].label, location=self.selected_location[0].label)
-        self.UpdateCurrentLocationMapsetNode()
+            self.changeGrassDb.emit(mapset=self.selected_mapset[0].label,
```

Comment:
Changing to location in the same or in a different db is the same, so changeGrassDb is not needed, just use changeLocation but add the grassdb parameter.

Reply:
How do you mean that? grassdb is not the parameter in changeLocation function, I need to add it but I do not know how. I sought in signal.py but I think you have meant it differently.

Reply:
Search for changeLocation.connect, that's how signal handlers are called.